src/ChatManager.cpp

```diff
@@ -699,6 +699,37 @@ void whoisCommand(std::string full, std::vector<std::string>& args, CNSocket* so
     ChatManager::sendServerMessage(sock, "[WHOIS] Instance: " + std::to_string(PLAYERID(npc->instanceID)));
 }
 
+void lairUnlock(std::string full, std::vector<std::string>& args, CNSocket* sock) {
+    Player* plr = PlayerManager::getPlayer(sock);
+    BaseNPC* npc = NPCManager::getNearestNPC(plr->viewableChunks, plr->x, plr->y, plr->z);
+    int taskID=-1;
+    int missionID = -1;
+    for (auto& pair : NPCManager::Warps) {
+        int currentNPCID = pair.second.npcID;
+        if (currentNPCID == npc->appearanceData.iNPCType) {
+            taskID = (int)pair.second.limitTaskID;
+            missionID = MissionManager::Tasks[taskID]->task["m_iHMissionID"];
+            break;
+        }
+        
+    }
+
+    if (missionID == -1 || taskID == -1)
+        return; // not a lair
+
+    INITSTRUCT(sP_FE2CL_REP_PC_SET_CURRENT_MISSION_ID, missionResp);
+    missionResp.iCurrentMissionID = missionID;
+    plr->CurrentMissionID = missionID;
+    sock->sendPacket((void*)&missionResp, P_FE2CL_REP_PC_SET_CURRENT_MISSION_ID, sizeof(sP_FE2CL_REP_PC_SET_CURRENT_MISSION_ID));
+
+
+    INITSTRUCT(sP_FE2CL_REP_PC_TASK_START_SUCC, taskResp);
+    MissionManager::startTask(plr, taskID, true);
```

Comment:
You shouldn't call `startTask` directly, because things like quest items won't get properly cleaned up.